mne/epochs.py

```diff
@@ -1672,6 +1673,9 @@ def _get_data(
         if (units is not None) and out:
             ch_factors = _get_ch_factors(self, units, picks)
 
+        if copy:
+            data = data.copy()
```

Comment:
I suggest moving this up about 25 lines, inside the `if` branch of the `if self.preload` conditional.  The `else` branch defines `data = np.empty(...)` so that will *never* be a view of the object's data and thus the `copy` will never be necessary.